src/auth/auth.ts

```diff
@@ -10,55 +18,112 @@ export class Auth extends FirebaseService {
+    if (payload.nbf && payload.nbf > Math.floor(Date.now() / 1000))
+      throw 'JWT token not yet valid';
+    if (payload.exp && payload.exp <= Math.floor(Date.now() / 1000))
+      throw 'JWT token expired';
     const jsonWebKey = await getPublicKey(kid);
-    if (alg !== 'RS256' || !jsonWebKey || payload.iss !== `https://securetoken.google.com/${this.settings.projectId}`) throw new Error('JWT invalid');
-    const key = await crypto.subtle.importKey('jwk', jsonWebKey, importAlgorithm, false, ['verify']);
-    const verified = await crypto.subtle.verify(importAlgorithm, key, Base64URL.parse(tokenParts[2]), (jwt as any)._utf8ToUint8Array(`${tokenParts[0]}.${tokenParts[1]}`));
+    if (
+      alg !== 'RS256' ||
+      !jsonWebKey ||
+      payload.iss !==
+        `https://securetoken.google.com/${this.settings.projectId}`
+    )
+      throw new Error('JWT invalid');
+    const key = await crypto.subtle.importKey(
+      'jwk',
+      jsonWebKey,
+      importAlgorithm,
+      false,
+      ['verify']
+    );
+    const verified = await crypto.subtle.verify(
+      importAlgorithm,
+      key,
+      Base64URL.parse(tokenParts[2]),
+      (jwt as any)._utf8ToUint8Array(`${tokenParts[0]}.${tokenParts[1]}`)
+    );
     if (!verified) throw new Error('JWT invalid');
     return payload;
   }
 
-  async signInWithEmailAndPassword(email: string, password: string): Promise<SignInResponse> {
+  async signInWithEmailAndPassword(
+    email: string,
+    password: string
+  ): Promise<SignInResponse> {
     email = email && email.toLowerCase();
     const data = { email, password, returnSecureToken };
-    const result: SignInFirebaseResponse = await this.userRequest('POST', 'accounts:signInWithPassword', data);
+    const result: SignInFirebaseResponse = await this.userRequest(
+      'POST',
+      'accounts:signInWithPassword',
+      data
+    );
     const tokens = convertSignInResponse(result);
     const user = await this.getUser(tokens.idToken);
     return { user, tokens };
   }
 
   // 0auth signing
-  async signInWithIdp(credentials: string, requestUri: string, returnIdpCredential = false): Promise<SignInResponse> {
-    const data = { postBody: credentials, requestUri, returnSecureToken, returnIdpCredential };
-    const result: SignInFirebaseResponse = await this.userRequest('POST', 'accounts:signInWithIdp', data);
+  async signInWithIdp(
+    credentials: string,
+    requestUri: string,
+    returnIdpCredential = false
+  ): Promise<SignInResponse> {
+    const data = {
+      postBody: credentials,
+      requestUri,
+      returnSecureToken,
+      returnIdpCredential,
+    };
+    const result: SignInFirebaseResponse = await this.userRequest(
+      'POST',
+      'accounts:signInWithIdp',
+      data
+    );
     const tokens = convertSignInResponse(result);
+
     const user = await this.getUser(tokens.idToken);
-    return { user, tokens };
+    return { user, tokens, isNewUser: result?.isNewUser };
```

Comment:
Need this for knowing if the google user is new or not for onetap/login with google.